CriteoPublisherSdk/Sources/Network/CR_ApiHandler.m

```diff
@@ -121,7 +121,8 @@ - (void)doCdbApiCall:(CR_CacheAdUnitArray *)adUnits
               context:(CRContextData *)contextData
         beforeCdbCall:(CR_BeforeCdbCall)beforeCdbCall
     completionHandler:(CR_CdbCompletionHandler)completionHandler {
-  CR_CacheAdUnitArray *requestAdUnits = [self filterRequestAdUnitsAndSetProgressFlags:adUnits];
+  CR_CacheAdUnitArray *filteredAdUnits = [self.integrationRegistry filterAdUnits:adUnits];
```

Comment:
This filter is done in a quite low level here. Isn't it possible to move it a little bit before.

I see one (weird) case which might provoke a bug:
- given CDB answering a bid
- given the SDK storing the bid in the cache as a rewarded ad unit (either because CDB wrongly answered or because of an adjacent bug in the SDK)
- when getting a next bid
- then bid in cache is retrieved
- then a rewarded bid is used while it's not usable

Of course we don't want the backend to answer crap, but we never know. For instance for native requests CDB was not always answering native bids before a bug was fixed.